rust/tools/authorization-logic/transparent_release_prototype/transparent_release.auth_logic

```diff
@@ -0,0 +1,276 @@
+// producing hashes instead of just one, so the trust assumptions would be less 
+// clear.)
+
+// The arguments to this program are:
+// - the name of the file to be measured (hash checked)
+// - the name of the principal for the file being measured
+
+// We will invoke this wrapper twice: once for the provenance file and once
+// for the binary generated by executing the steps in the provenance file
+
+// For the provenance file, the inputs are:
+//    file name --> "provenance_file.toml"
+//    principal name --> "ProvenanceFile_OakFunctionsLoader"
+// The output looks like:
+"Sha256Wrapper" says {
+    measured_hash("ProvenanceFile_OakFunctionsLoader",
+      "6de656f11599c11548519af6b7385504ed845a4ff6600745cb3d9e174b8956af").
+}
+
+// For the binary, the inputs are:
+//    file name --> "oak_functions_loader.exe" (I don't have an example binary)
+//    principal name --> "Binary_OakFunctionsLoader"
+"Sha256Wrapper" says {
+    measured_hash("Binary_OakFunctionsLoader",
+      "cb255dbbea86b128379995ff7dc805fd026e2d21c997a8df688ba9e5cb98e1c1").
+}
+
+//-----------------------------------------------------------------------------
+// Unix Time Wrapper
+//-----------------------------------------------------------------------------
+// We need a wrapper that produces the real time in unix epoch seconds. A 
+// command for doing this on UNIX systems is `date %s`. Assuming the time
+// that this command is run is 2021-10-21, 19:14:58 the output from running
+// this command is 1634843698.
+//
+// This wrapper is also a standalone binary. It takes no inputs and outputs
+// authorization logic code that looks like this:
+
+"UnixEpochTime" says {
+    RealTimeIs("1634843698").
+}
+
+//-----------------------------------------------------------------------------
+// Rekor Log Check Wrapper
+//-----------------------------------------------------------------------------
+// To check if an endorsement file really is a valid rekor log entry,
+// the rekor CLI is called.
+//
+// Inputs to wrapper:
+//     - name of app (for generating principal names)
+//     - public key path of product team
+//     - path to endorsement file
+//     (- optionally: the product team could supply the UUID of the
+//     specific record for this version of the endorsement file
+//     and the endorsement file signature to skip the for loop)
+// Internally:
+//      - the wrapper calls `rekor-cli search --public-key=<pubkey> 
+//              --pki-format=x509` 
+//       to get all the UUIDs for records with this key
+//      - for each UUID:
+//            - call `rekor-cli get --uuid=<UUID>` returning
+//            someting that looks like `rekor_log_cli_output.txt`
+//            - fetch `signature.content` and verify this signature
+//            with the signature of the endorsement file from
+//            the prod team. Succeed and produce auth logic output if
+//            verification succeeds
+//      - if loop terminates without success, throw error
```

Comment:
I don't think even the wrapper should connect to Rekor directly, it should accept the pre-downloaded entries as parameters. The entire generation of this file should not require any wrapper to even have any connection to the internet. The only additional input is perhaps the Rekor public key (also pre-downloaded) in order to validate the signature.

Anyways I don't think this changes anything about this example, but I wanted to clarify expectations.

Reply:
Sounds good! :)